### PR TITLE
Added validation for scheduled workflow during registration

### PR DESCRIPTION
**Pull request by pmahindrakar-oss**
Signed-off-by: Prafulla Mahindrakar <prafulla.mahindrakar@gmail.com>

# TL;DR
Adding validation during registration for scheduled workflows.

If the number of params is greater than the fixed inputs then it raises an error.
In normal case the scheduled workflow has atmost one param kickofTimeArg.

Tested :
Failure case
```
 --------------------------------------------------------------------------------------------------------------------- --------- --------------------------------------------------- 
| /Users/praful/flytesnacks/cookbook/core/_pb_output/120_my_cron_scheduled_lp_3.pb                                    | Failed  | Error registering file due to rpc error: code =   |
|                                                                                                                     |         | InvalidArgument desc = launch plan with different |
|                                                                                                                     |         | structure already exists with id                  |
|                                                                                                                     |         | resource_type:LAUNCH_PLAN project:"flytesnacks"   |
|                                                                                                                     |         | domain:"development" name:"my_cron_scheduled_lp"  |
|                                                                                                                     |         | version:"v7"                                      |
 --------------------------------------------------------------------------------------------------------------------- --------- --------------------------------------------------- 
```

Success case:

```
 --------------------------------------------------------------------------------------------------------------------- --------- ------------------------------ 
| /Users/praful/flytesnacks/cookbook/core/_pb_output/120_my_cron_scheduled_lp_3.pb                                    | Success | Successfully registered file |
 --------------------------------------------------------------------------------------------------------------------- --------- ------------------------------ 
```

## Type
- [X] Bug Fix
- [ ] Feature
- [ ] Plugin

## Are all requirements met?

- [ ] Code completed
- [ ] Smoke tested
- [ ] Unit tests added
- [ ] Code documentation added
- [ ] Any pending items have an associated Issue

## Complete description
_How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/flyteorg/flyte/issues/1817

## Follow-up issue
_NA_

### CODE REVIEW

**Review by EngHabu — DISMISSED** [review]

File: cmd/register/register_util.go [hunk 1]
```diff
@@ -276,14 +307,18 @@ func hydrateLaunchPlanSpec(lpSpec *admin.LaunchPlanSpec) {
 			OutputLocationPrefix: rconfig.DefaultFilesConfig.OutputLocationPrefix,
```
Comment (evalsocket) [inline cmd/register/register_util.go:307]: Need to refactor `rconfig.DefaultFilesConfig.OutputLocationPrefix`. We can't use directly as per new change

File: cmd/register/register_util.go [hunk 2]
```diff
@@ -276,14 +307,18 @@ func hydrateLaunchPlanSpec(lpSpec *admin.LaunchPlanSpec) {
...
 }
 
 func hydrateSpec(message proto.Message, sourceCode string) error {
 	switch v := message.(type) {
 	case *admin.LaunchPlan:
 		launchPlan := message.(*admin.LaunchPlan)
 		hydrateIdentifier(launchPlan.Spec.WorkflowId)
-		hydrateLaunchPlanSpec(launchPlan.Spec)
+		err := hydrateLaunchPlanSpec(launchPlan.Spec)
+		if err != nil {
```
Comment (evalsocket) [inline cmd/register/register_util.go:319]: Make it one liner 
```
if err := hydrateLaunchPlanSpec(launchPlan.Spec); err != nil {
    return err
}
```

Comment (pmahindrakar-oss) [reply cmd/register/register_util.go]: Done

File: cmd/register/register_util.go (hunk 1 above)
Comment (pmahindrakar-oss) [reply cmd/register/register_util.go]: Done.Rebased

**Review by evalsocket — APPROVED** [review]
